src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -1436,6 +1428,22 @@ private void CreateMutatedSpeciesCopy(Species species)
         }
     }
 
+    /// <summary>
+    ///   Calculates the remaining MP from the action history
+    /// </summary>
+    /// <returns>The remaining MP</returns>
+    private int CalculateMutationPointsLeft()
+    {
+        if (FreeBuilding || CheatManager.InfiniteMP)
+            return Constants.BASE_MUTATION_POINTS;
+
+        mutationPointsCache = history.CalculateMutationPointsLeft().Clamp(0, Constants.BASE_MUTATION_POINTS);
```

Comment:
It doesn't have to be there. I've added it to make sure the result is clamped.

Reply:
I would rather fail loudly here. If we end up in a situation where `CalculateMutationPointsLeft()` returns a negative number, something has already gone wrong, and I'd like to know sooner rather than later.